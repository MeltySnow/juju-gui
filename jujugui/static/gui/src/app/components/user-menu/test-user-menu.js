/* Copyright (C) 2017 Canonical Ltd. */
'use strict';

const React = require('react');
const enzyme = require('enzyme');

const UserMenu = require('./user-menu');

describe('UserMenu', () => {

  const loginLink = <div className="login"></div>;
  const logoutLink = <div className="logout"></div>;

  const renderComponent = (options = {}) => enzyme.shallow(
    <UserMenu
      LogoutLink={options.LogoutLink || logoutLink}
      navigateUserProfile={options.navigateUserProfile || sinon.stub()}
      showHelp={sinon.stub()}
      showLogin={options.showLogin !== undefined ? options.showLogin : true}
      USSOLoginLink={
        options.USSOLoginLink !== undefined ?
          options.USSOLoginLink : loginLink} />
  );

  it('can render', () => {
    const wrapper = renderComponent({
      userIsAuthenticated: false
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the login link if supplied', () => {
    const wrapper = renderComponent({
      userIsAuthenticated: false
    });
    assert.deepEqual(wrapper.find('ButtonDropdown').prop('icon'), loginLink);
  });

  it('renders a user icon when no login link is supplied', () => {
    const wrapper = renderComponent({
      USSOLoginLink: null
    });
    assert.equal(wrapper.find('ButtonDropdown').prop('icon'), 'user_16');
  });

  it('navigates to user profile when clicked', () => {
    const navigateUserProfile = sinon.stub();
    const wrapper = renderComponent({navigateUserProfile});
    wrapper.find('ButtonDropdown').prop('listItems')[0].action();
    assert.equal(navigateUserProfile.callCount, 1);
  });

});
