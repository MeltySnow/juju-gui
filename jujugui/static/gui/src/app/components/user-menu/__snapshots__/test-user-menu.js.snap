// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenu can render 1`] = `
<div>
  <ButtonDropdown
    classes={
      Array [
        "user-menu",
      ]
    }
    disableDropdown={true}
    icon={
      <div
        className="login"
      />
    }
    listItems={
      Array [
        Object {
          "action": [Function],
          "label": "Profile",
        },
        Object {
          "action": [Function],
          "label": "GUI help",
        },
        Object {
          "element": <div
            className="logout"
          />,
        },
      ]
    }
    tooltip=""
  />
</div>
`;
